src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -140,21 +144,25 @@ object Bams extends LazyLogging {
     * a queryname sort, merely a query grouping.
     *
     * @param in a SamReader from which to consume records
-    * @param maxInMemory the maximum number of records to keep and sort in memory if sorting is needed
+    * @param maxInMemory optionally the maximum number of records to keep and sort in memory if sorting is needed
     * @param tmpDir an optional temp directory to use for temporary sorting files if needed
+    * @param tagsToRemove an optional list of SAM tags to remove after reading a record.
     * @return an Iterator with reads from the same query grouped together
     */
   def queryGroupedIterator(in: SamReader,
-                           maxInMemory: Int = 1e6.toInt,
-                           tmpDir: Option[DirPath] = None): BetterBufferedIterator[SAMRecord] = {
+                           maxInMemory: Option[Int] = None,
+                           tmpDir: Option[DirPath] = None,
+                           tagsToRemove: Seq[String] = Seq.empty): BetterBufferedIterator[SAMRecord] = {
```

Comment:
This is super-specific, and feels like a response to a particular problem. Instead of doing this, how about creating versions of these methods that take in `SAMFileHeader` and `Iterator[SAMRecord]` (that the current versions could delegate to), thus allowing you to `filter`, `map`, etc. on the iterators to your heart's content before you pass them in rather than building in one specific kind of transform here?